### ISSUE TITLE
Cross reference numbering doesn't work as expected

### ISSUE THREAD

**Issue by sinaatalay** [tobiasah/mkdocs-caption#19]
Firstly, thank you very much for the project.

I have a small error. Here is a sample Markdown to reproduce it:

```md
This should be Figure 2: [](#test2) (but it's Figure 1).


![this is a caption](test.png){#test}

![this is a caption](test.png){#test2}
```

![image](https://github.com/tobiasah/mkdocs-caption/assets/79940989/b62286c3-979a-4a66-a041-b153dcb2e31c)

**Comment by tobiasah** [comment]
thanks for reporting this. Its a bug I introduced with the latest change. 

I fixed it with #20 . Can you check if this fixes the problem? (By installing python package directly from the branch `tobiasa/19`)

**Comment by sinaatalay** [comment]
Thank you! I tested it, it's working.
